### ISSUE TITLE
Model for OneAccess TDRE

### ISSUE THREAD

**Issue by roelpa** [ytti/oxidized#1693]
Hello

Would anyone be able to point me in the right direction. I would like to use oxidized to take configuration backup from OneAccess TDRE devices.

I've created a new model for TDRE, but it's failing on the login prompt.
When logging into the box with SSH, this is the prompt:

[oxidized@LibreNMS oxidized]$ ssh ops@10.246.0.21
ops@10.246.0.21's password:

Login successful -- CLI active -- connecting with device
>
/router1645:"Edit Configuration"
>

In the logs, I can see the following:

W, [2019-01-24T14:30:03.621136 #16066]  WARN -- : 10.246.0.21 raised Oxidized::PromptUndetect with msg "
Login successful -- CLI active -- connecting with device
>
/router1645:"Edit Configuration"
> not matching configured prompt (?-mix:^([-\w.\/:?\[\]()]+:\s?)$)"
D, [2019-01-24T14:30:03.621360 #16066] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for 10.246.0.21
D, [2019-01-24T14:30:03.621486 #16066] DEBUG -- : lib/oxidized/job.rb: Config fetched for 10.246.0.21 at 2019-01-24 13:30:03 UTC
W, [2019-01-24T14:30:04.627250 #16066]  WARN -- : /10.246.0.21 status no_connection, retry attempt 1
D, [2019-01-24T14:30:04.627376 #16066] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 7 of 11
D, [2019-01-24T14:30:04.627455 #16066] DEBUG -- : lib/oxidized/worker.rb: Added /10.246.0.21 to the job queue
D, [2019-01-24T14:30:04.627486 #16066] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2019-01-24T14:30:04.632404 #16066] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for 10.246.0.21 at 2019-01-24 13:30:04 UTC
D, [2019-01-24T14:30:04.632683 #16066] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to 10.246.0.21
D, [2019-01-24T14:30:04.632842 #16066] DEBUG -- : AUTH METHODS::["none", "publickey", "password", "keyboard-interactive"]

Any suggestions on how the model needs to be altered?

class TDRE < Oxidized::Model
  # OneAccess TDRE

  comment '# '

  prompt /^([-\w.\/:?\[\]()]+:\s?)$/

  cmd 'GET -f' do |cfg|
    cfg
  end

  cmd :all do |cfg|
    cfg.cut_both
  end

  cfg :ssh do
    pre_logout 'DISCONNECT'
  end
end

regards
Roel

**Comment by starrsl** [comment]
I found the following worked : 

    prompt /^((\/\w+:"[^"]+")|(.* -- connecting with device)\n>)/

**Comment by starrsl** [comment]
I do not as i am in a similar situation in that i am attempting to build one at the moment. 

I however have noticed an odd issue with the TDRE in that a basic config similar to what you posted above will work under telnet but not under SSH. If you have the potential to test under telnet vs ssh could you report back with your results?

**Comment by wk** [comment]
@roelpa #1701 has now been merged into master; if you could test this as well and report your outcome with telnet and ssh so that @starrsl can do final tweaks to the model if necessary, that would be great.

Otherwise, I'm making this issue as `user_pending` so it'd autoclose if no further feedback is received.

**Comment by roelpa** [comment]
The model seems to work well for SSH.
I'll try to test it with Telnet as well, but we generally disable telnet by default.
